homeassistant/components/hive/alarm_control_panel.py

```diff
@@ -0,0 +1,103 @@
+"""Support for the Hive alarm."""
+from datetime import timedelta
+
+from homeassistant.components.alarm_control_panel import AlarmControlPanelEntity
+from homeassistant.components.alarm_control_panel.const import (
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_NIGHT,
+)
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_TRIGGERED,
+)
+
+from . import HiveEntity
+from .const import DOMAIN
+
+ICON = "mdi:security"
+PARALLEL_UPDATES = 0
+SCAN_INTERVAL = timedelta(seconds=15)
+HIVETOHA = {
+    "home": STATE_ALARM_DISARMED,
+    "asleep": STATE_ALARM_ARMED_NIGHT,
+    "away": STATE_ALARM_ARMED_AWAY,
+}
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Hive thermostat based on a config entry."""
+
+    hive = hass.data[DOMAIN][entry.entry_id]
+    devices = hive.session.deviceList.get("alarm_control_panel")
+    entities = []
+    if devices:
+        for dev in devices:
+            entities.append(HiveAlarmControlPanelEntity(hive, dev))
+    async_add_entities(entities, True)
+
+
+class HiveAlarmControlPanelEntity(HiveEntity, AlarmControlPanelEntity):
+    """Representation of a Hive alarm."""
+
+    @property
+    def unique_id(self):
+        """Return unique ID of entity."""
+        return self._unique_id
+
+    @property
+    def device_info(self):
+        """Return device information."""
+        return {
```

Comment:
I realize this PR has been sitting for months -- now probably worth adding `DeviceInfo` here given that style was changed.